### PR TITLE
Path expand fixes

### PR DESCRIPTION
**Pull request by kubouch**
Adding extra functionality to make `path expand` more intuitive & flexible to use.

Needs better examples & tests. Also, see the spans bug below.

### CODE REVIEW

**Comment by kubouch** [comment]
There is a strange error which I think points to a bug with how Spans are calculated:
```
thread 'commands::path::expand::tests::examples_work_as_expected' panicked at 'example command produced unexpected result.
 command: '/home/joe/foo/../bar' | path expand
 expected: Value {
    value: Primitive(
        String(
            "/home/joe/bar",
        ),
    ),
    tag: Tag {
        anchor: None,
        span: Span {
            start: 0,
            end: 13,
        },
    },
}
actual: Value {
    value: Primitive(
        FilePath(
            "/home/joe/bar",
        ),
    ),
    tag: Tag {
        anchor: None,
        span: Span {
            start: 0,
            end: 22,
        },
    },
}', crates/nu-command/src/examples.rs:135:21
```
The "actual" value has the span calculated before the `..` path expansion.

**Comment by jonathandturner** [comment]
@kubouch - it's definitely possible it's some logic we need to add to the test harness

tagging @andrasio who might know

**Comment by andrasio** [comment]
@kubouch Correct me if I'm wrong but, why do you think span `(0,22)` for `'/home/joe/foo/../bar' | path expand` is not appropriate? If your command `path expand` fails and you'd like to report an error, you'd need to underline in red `'/home/joe/foo/../bar'` and the span you are seeing seems correct for that case.

```
Span::new(0.20).slice("'/home/joe/foo/../bar' | path expand") => '/home/joe/foo/../bar'
```

**Comment by andrasio** [comment]
Perhaps try in your expectation being explicit with the tag and not just `Value::from(path_expanded)`

**Comment by kubouch** [comment]
@andrasio Yes, that makes total sense! The span should refer to the input value, not the output. I added the span manually to the tests.

**Comment by kubouch** [comment]
I was quite busy last week but here it is, finally. Now, `path expand` can be used to expand `~`, `.` and `..` in relative and non-existent paths.

File: crates/nu-engine/src/filesystem/path.rs [hunk 1]
```diff
@@ -1,6 +1,25 @@
 use std::io;
 use std::path::{Component, Path, PathBuf};
 
+pub fn resolve_dots<P>(path: P) -> PathBuf
```
Comment (jonathandturner) [inline crates/nu-engine/src/filesystem/path.rs:4]: Feels like this is something we'd want to use for all our paths that we show to the user

Comment (jonathandturner) [reply crates/nu-engine/src/filesystem/path.rs:4]: (or something like it)

**Comment by jonathandturner** [comment]
Looks good!

**Pull request merged**